yt/utilities/lib/fp_utils.pxd

```diff
@@ -58,3 +58,7 @@ cdef inline _ensure_code(arr):
             return arr
         arr.convert_to_units("code_length")
     return arr
+
+ctypedef fused any_float:
```

Comment:
While we're at it, any reason not to include float16 and float128 ?

Reply:
Only that I don't think we use them anywhere?  Maybe in a future revision, if we end up having a use case that we can trust with them.

Reply:
Sounds fair. Thanks !

Reply:
Is there a use case for also applying the fuse type to e.g. `pixelize_sph_kernel_arbitrary_grid` et al.?

Reply:
Yes -- I'll add some of these once the tests pass and I know that I'm on the right track.  :)

Reply:
```
ValueError: Buffer dtype mismatch, expected 'float32_t' but got 'double'
```

Reply:
Is this still an issue?

Reply:
I think this one is OK to resolve since tests are passing now, ya?

@neutrinoceros ?

Reply:
sure. And unless I'm missing something, seems to me that's the last hanging thread, so maybe you can merge now ?

Reply:
Agreed! I think we're ready!